### PR TITLE
[#405] Infrastructure / Architecture Documentation

### PR DESCRIPTION
**Pull request by lancelafontaine**
[#405]

- [x] Writes documentation for infrastructure and architecture
- [x] Renders the documentation website with mdBook

### CODE REVIEW

File: src/README.md [hunk 1]
```diff
@@ -1,23 +1,53 @@
...
+
+Once a service's tests have completed running, the test line code covereage is uploaded to [CodeCov.io](https://codecov.io/gh/Cyberjusticelab/JusticeAI). This service ensures that we're always maintaining a reasonable number of tests for our application over time. This check is more informational, and is not required to pass for a build to be merged.
+
+We do currently have some sort of continuous deployment onto a VPS provided to us by UdeM at `capstone.cyberjustice.ca`. It's currently configured to deploy after the end of a successful Travis build, but with the parallelization, it often fails and is extremely error prone. We've begun investigating standing up a webhook notification server on `capstone.cyberjustice.ca` that GitHub can push a notification to once a PR is merged, but it has not been flawless either.
```
Comment (ghost) [inline src/README.md]: Open source community won't have access to this, no point in mentioning it.

Especialy the "it has not been flawless either" point.

Comment (lancelafontaine) [reply src/README.md]: It was meant describe the current situation for future maintainers.

File: src/README.md [hunk 2]
```diff
@@ -1,23 +1,53 @@
...
+
+We are using the default PostgreSQL docker image as a persistent data store. At the moment, we do not keep track of database migrations, and the database is wiped and rebuilt on every deployment. This can also be done manually on your local machine with `./cjl reset-db`. The services that require DB access run an `init.py` script during build time that constructs the database models.
+
+Due to this, we've noticed that sometimes the database takes ~30 sec to create the models at runtime and be ready to accept connections, but the application services have thrown an error due to the requirement of creating a database connection. If you simply `./cjl down && ./cjl up`, this problem goes away, but it may be worth having the application servers stall and wait as they perform a health check on the database before attempting to create a database connections.
```
Comment (ghost) [inline src/README.md]: This should be in the Troubleshoot.md

File: src/README.md [hunk 3]
```diff
@@ -1,23 +1,53 @@
...
+
+## Future Improvements
```
Comment (ghost) [inline src/README.md]: Unsure if the MD is the best for the roadmap especially considering we won't develop this application anymore.

Make github issues about them instead.

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — CHANGES_REQUESTED** [review]
IMO some changes to be made, quick ones.

Optional changes.

File: src/README.md (hunk 1 above)
Comment (ghost) [reply src/README.md]: They know the situation already. They're being handed the project.

This MD has to intersect maintainers + open source community IMO.

**Review by ghost — APPROVED** [review]

**Review by arekmano — APPROVED** [review]

**Pull request merged**